arviz/plots/rankplot.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+
+from ..data import convert_to_dataset
+from .plot_utils import (
+    _scale_fig_size,
+    xarray_var_iter,
+    default_grid,
+    _create_axes_grid,
+    make_label,
+)
+from ..utils import _var_names
+
+
+def plot_rank(data, var_names=None, coords=None, bins="auto", figsize=None, axes=None):
+
+    posterior_data = convert_to_dataset(data, group="posterior")
+    if coords is not None:
+        posterior_data = posterior_data.sel(**coords)
+    var_names = _var_names(var_names, posterior_data)
+    plotters = list(xarray_var_iter(posterior_data, var_names=var_names, combined=True))
+
+    if axes is None:
+        rows, cols = default_grid(len(plotters))
+
+        figsize, ax_labelsize, titlesize, _, _, _ = _scale_fig_size(
+            figsize, None, rows=rows, cols=cols
+        )
+        _, axes = _create_axes_grid(len(plotters), rows, cols, figsize=figsize, squeeze=False)
+
+    for ax, (var_name, selection, var_data) in zip(axes.ravel(), plotters):
+        ranks = var_data.argsort(axis=None).reshape(*var_data.shape)
+        all_counts = []
+        for row in ranks:
+            # If `bins` was auto, it gets overwritten here, and carried forward
+            # using the `range` argument makes sure the bins cover all the data
+            counts, bins = np.histogram(row, bins=bins, range=(0, ranks.size))
```

Comment:
This is not what we want. It will not return the consistent amount of bins (tested with out example data)

We want to make bins only based on N, not based on data (what "auto" will give us).

Either we need to iterate all chains and take the maximum bin_count and use that or we can use our own "sturges" here (what R uses)

    def sturges(ary):
        return int(np.ceil(np.log2(ary.size)) + 1)

Reply:
I think it would be even better if we would go with 

    def double_sturges(ary):
        return int(np.ceil(2 * np.log2(ary.size)) + 1)

Reply:
i guess this is easy for us, since `ary.size` will be just `data.posterior.draw.size`.